core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java

```diff
@@ -36,12 +38,34 @@
 public class RewriteFileGroup {
   private final FileGroupInfo info;
   private final List<FileScanTask> fileScanTasks;
+  private final Set<DeleteFile> removedDeletes;
 
   private Set<DataFile> addedFiles = Collections.emptySet();
 
-  public RewriteFileGroup(FileGroupInfo info, List<FileScanTask> fileScanTasks) {
+  public RewriteFileGroup(FileGroupInfo info, List<FileScanTask> fileScanTasks,
+                          boolean removePartitionDeletes, boolean removeGlobalDeletes) {
     this.info = info;
     this.fileScanTasks = fileScanTasks;
+    this.removedDeletes = Sets.newHashSet();
+    for (FileScanTask task : fileScanTasks) {
+      if (task.spec().isUnpartitioned()) {
+        if (removePartitionDeletes) {
```

Comment:
If the task.spec().isUnpartitioned() == true, should we handle globalDelete files as well..
from the code, it seems that only the partitionDelete files are considered.